### PR TITLE
[googlemaps] Hotfix MouseEvent

### LINKED ISSUE #50253: [@types/googlemaps] - Namespace 'google.maps' has no exported member 'MouseEvent'
[ x ] I tried using the @types/googlemaps package and had problems.
[ x ] I understand that @types package versions do not follow SemVer.
[ x ] I have a question that is inappropriate for StackOverflow. (Please ask any appropriate questions there).
[ x ] Mention the authors (see Definitions by: in index.d.ts) so they can respond.
Authors: @demensky @cgwrench 

I have an angular 9 project and, after an `npm install`, the build process it stops with the following error:
```
ERROR in node_modules/@angular/google-maps/google-map/google-map.d.ts:68:38 - error TS2694: Namespace 'google.maps' has no exported member 'MouseEvent'.

68     mapClick: Observable<google.maps.MouseEvent | google.maps.IconMouseEvent>;
```
This occurs for the version 3.43.0 of `@types/googlemaps`, with `@angular/google-maps` 9.2.4 version.
A temporary fix for the problem is to add as devDependencies `@types/googlemaps` with an erlier version (3.39.14)

### PR DESCRIPTION
**Pull request by demensky**
Fixes #50253. 

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://developers.google.com/maps/documentation/javascript/releases#2020-12-09
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/googlemaps/reference/map.d.ts [hunk 1]
```diff
@@ -1,3 +1,9 @@
+// TODO: #50253 remove after that most libraries will go to MapMouseEvent
+/**
+ * @internal
+ */
+type GoogleMapsNativeMouseEvent = MouseEvent;
```
Comment (demensky) [inline types/googlemaps/reference/map.d.ts:5]: This is the only way I know how to use a global type whose name is the same as the name from the local namespace.

**Comment by demensky** [comment]
It also turned out that not all files had interfaces renamed in #50188 correctly. @kenjdavidson please see this PR too.

**Comment by kenjdavidson** [comment]
Oh man! Sorry I missed those. Thanks for the backup! Keep safe.

File: types/googlemaps/test/reference/marker-tests.ts [hunk 2]
```diff
@@ -50,7 +50,8 @@ marker.getOpacity(); // $ExpectType number | null | undefined
 
 marker.getPosition(); // $ExpectType LatLng | null | undefined
 
-marker.getShape(); // $ExpectType MarkerShapeCircle | MarkerShapeRect | MarkerShapePoly | null | undefined
+// TODO: In typescript 4.2, ExpectType does not expand the type.
+// marker.getShape(); // $ExpectType MarkerShapeCircle | MarkerShapeRect | MarkerShapePoly | null | undefined
```
Comment (demensky) [inline types/googlemaps/test/reference/marker-tests.ts:54]: In TypeScript 4.2, `$ExpectType` behaves differently.

**Review by simonhaenisch — APPROVED** [review]
looks good to me, the workaround with the alias for the global type seems fine.

**Review by simonhaenisch — APPROVED** [review]

**Comment by demensky** [comment]
Ready to merge

**Pull request merged**